warning: ‘strncpy’ when using -march=native

When trying to squeeze it a bit with an -march=native compile on an AMD Zen 3 CPU there's a warning:

```
$ git log | head -3
From https://github.com/labwc/labwc
 * branch            master     -> FETCH_HEAD
Already up to date.
commit 5cc6a5b3e1f4b1a5b82869910722fb71b2c08348
Author: Johan Malm <jgm323@gmail.com>
Date:   Sat Dec 9 19:49:23 2023 +0000



$ gcc -c -Q -march=native --help=target | grep 'march=  '
  -march=                         znver3



$ export CFLAGS="-march=native -O2 -pipe"
$ export CXXFLAGS="${CFLAGS}"



$ meson subprojects update --reset
Updating seatd...
  -> Not used.
Updating wlroots...
  -> fe53ec69 (HEAD -> 0.17, origin/0.17) tinywl: fix wlroots dependency constraint in Makefile [Simon Ser]



$ meson setup --wipe --reconfigure ./build >/dev/null



$ meson compile -C ./build
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: /usr/bin/ninja -C /tmp/labwc/build
ninja: Entering directory `/tmp/labwc/build'
[204/433] Compiling C object labwc.p/src_output.c.o
../src/output.c: In function ‘output_add_virtual’:
../src/output.c:772:17: warning: ‘strncpy’ specified bound 4096 equals destination size [-Wstringop-truncation]
  772 |                 strncpy(server->headless.pending_output_name, output_name,
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  773 |                                 sizeof(server->headless.pending_output_name));
      |                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[433/433] Linking target labwc
```

It's just a warning but looks like that place can be improved.

https://stackoverflow.com/questions/50198319/gcc-8-wstringop-truncation-what-is-the-good-practice/58237928#58237928

--- a/src/output.c
+++ b/src/output.c
@@ -769,8 +769,8 @@ output_add_virtual(struct server *server, const char *output_name)
 				return;
 			}
 		}
-		strncpy(server->headless.pending_output_name, output_name,
-				sizeof(server->headless.pending_output_name));
+		snprintf(server->headless.pending_output_name,
+			sizeof(server->headless.pending_output_name), "%s", output_name);
 	} else {
 		server->headless.pending_output_name[0] = '\0';
 	}

